src/main/kotlin/org/rust/lang/refactoring/RustNameSuggestions.kt

```diff
@@ -0,0 +1,101 @@
+package org.rust.lang.refactoring
+
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.codeStyle.NameUtil
+import com.intellij.psi.util.PsiTreeUtil
+import org.rust.ide.navigation.goto.RustSymbolNavigationContributor
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.util.childOfType
+import org.rust.lang.core.psi.util.parentOfType
+import org.rust.lang.core.types.util.resolvedType
+import java.util.*
+
+/**
+ * This suggests names for an expression about to be bound to a local variable
+ *
+ * If the type is resolved suggest the first letter of the type name.
+ * If its an argument to a function call, suggest the name of the argument in the function definition.
+ * If its a function call suggest the name of the function and the enclosing name space (if any).
+ * If a name is already bound in the local scope do not suggest it.
+ */
+fun suggestedNames(project: Project, expr: PsiElement): LinkedHashSet<String> {
```

Comment:
Maybe:
```kotlin
fun PsiElement.suggestNames()
```

?

You should be able to get project using `PsiElement#getProject()` method.